### DISCUSSION TITLE
Additional events for login actions

### DISCUSSION THREAD

**Discussion by lukeyouell** [craftcms/cms#8587] [Ideas]
There was a discussion on Slack on the topic of being able to prevent an email being sent after users login with `passwordResetRequired` set to `1`. One solution for this would be to use an `AFTER_LOGIN` event of sorts to perform the desired action before setting `passwordResetRequired` back to `0` before any email is sent. The only issue being that this particular event doesn't exist.

Would it be possible to have the one or more of the following events made available?

- Before login
- After successful login
- After failed login

Cheers!

**Comment by ChristopherDosin** [comment]
+1
